### PR TITLE
Add standalone executable for Servo server node

### PR DESCRIPTION
**Pull request by schornakj**
### Description

Previously the servo_server node was only available as a component node. This adds a new executable to run the servo_server as a standalone node.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/moveit_servo/src/servo_server_node.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+#include <moveit_servo/servo_server.h>
+
+int main(int argc, char* argv[])
+{
+  rclcpp::init(argc, argv);
+
+  rclcpp::NodeOptions options;
+  options.automatically_declare_parameters_from_overrides(true);
```
Comment (tylerjw) [inline moveit_ros/moveit_servo/src/servo_server_node.cpp:47]: In the future I hope we can get away from this by declaring all the parameters in a better way with nicer validation.  I'm working on that.

**Review by tylerjw — APPROVED** [review]
Looks really good and useful.  Thank you.  Only thing I'd add is an example launch file for it but it is so similar to the component node I'm not sure that's worth it.  I hope this is helpful in debugging the issues you've seen with component nodes.

File: moveit_ros/moveit_servo/src/servo_server_node.cpp (hunk 1 above)
Comment (AndyZe) [reply moveit_ros/moveit_servo/src/servo_server_node.cpp:47]: TBH I'm still fuzzy on the finer differences between component nodes and standalone nodes.

The PR looks good. I'll prob give it a thumbs up after testing.

**Comment by AndyZe** [comment]
@schornakj do you have a launch file to test with? My simple modification of `moveit2_tutorials/docs/realtime_servo/launch/servo_teleop.launch.py` did not work. I'm guessing the parameters were loaded in the wrong namespace.

[servo_teleop.launch.py.txt](https://github.com/ros-planning/moveit2/files/7028574/servo_teleop.launch.py.txt)

**Comment by schornakj** [comment]
> @schornakj do you have a launch file to test with? My simple modification of `moveit2_tutorials/docs/realtime_servo/launch/servo_teleop.launch.py` did not work. I'm guessing the parameters were loaded in the wrong namespace.
> 
> [servo_teleop.launch.py.txt](https://github.com/ros-planning/moveit2/files/7028574/servo_teleop.launch.py.txt)

I'll add a standalone launch file for this equivalent to the one we provide for the Servo component node.

**Review by henningkayser — APPROVED** [review]
Looks good to me. In general, would it be bad practice to add a main function to source files of component nodes directly? Say, add it to [servo_server.cpp](https://github.com/ros-planning/moveit2/blob/main/moveit_ros/moveit_servo/src/servo_server.cpp#L167) below registering the component node, or alternatively add the RCLCPP_COMPONENTS_REGISTER_NODE macro to `servo_server_node.cpp`? Also, the source file naming is really confusing, we have `servo.cpp` (main servo class), `servo_server.cpp` (node implementation) and `servo_server_node.cpp` which is the now added main function to run ServoServer as node executable. Out of scope of this PR, but we should clear that up.

**Comment by tylerjw** [comment]
> Agree that the naming is starting to be confusing.

I created a PR because I feel the same way: https://github.com/ros-planning/moveit2/pull/649

**Review by AndyZe — CHANGES_REQUESTED** [review]
blocking merge until I can test it

File: moveit_ros/moveit_servo/launch/servo_example.launch.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    for controller in ["panda_arm_controller", "joint_state_broadcaster"]:
+        load_controllers += [
+            ExecuteProcess(
+                cmd=["ros2 run controller_manager spawner.py {}".format(controller)],
+                shell=True,
+                output="screen",
+            )
+        ]
+
+    # Component nodes for tf and Servo
```
Comment (AndyZe) [inline moveit_ros/moveit_servo/launch/servo_example.launch.py]:
```suggestion
    # Component nodes for tf
    # Servo can also be launched as a component but this launch file uses a standalone node
```

**Comment by AndyZe** [comment]
@schornakj I updated the example launch file to:

- launch RViz
- include a commented example of launching Servo as a component, so that documentation is all in one place

I think the tutorial can now use this launch file rather than a completely separate one  :+1:

Let me know if that bugs you for some reason and we can revert the commits.

**Review by AndyZe — APPROVED** [review]

**Review by AndyZe — APPROVED** [review]

**Pull request merged**